tests/ha/control_pod/test_control_pod_soft_failures.py

```diff
@@ -0,0 +1,617 @@
+                'output': bkt_output}
+        thread2 = threading.Thread(target=self.ha_obj.iam_bucket_cruds, args=(event,), kwargs=args)
+        thread2.daemon = True  # Daemonize thread
+        thread1.start()
+        thread2.start()
+        LOGGER.info("Step 2: Successfully started IAM user and bucket creation/deletion in "
+                    "background")
+        LOGGER.info("Waiting for %s sec for background operations to start",
+                    HA_CFG["common_params"]["10sec_delay"])
+        time.sleep(HA_CFG["common_params"]["10sec_delay"])
+
+        LOGGER.info("Step 3: Create soft failures")
+        event.set()
+        for pod in self.ctrl_pod_list:
+            LOGGER.info("Creating soft failure on pod %s", pod)
+            resp = self.node_master_list[0].kill_process_in_container(
+                pod_name=pod, container_name=self.container_name, pid=self.pid, safe_kill=True)
+            LOGGER.info("Response: %s", resp)
+        LOGGER.info("Step 3: Created soft failures")
+
+        LOGGER.info("Waiting for %s time for all control pod containers to start",
+                    self.min_proc_restart_delay)
+        time.sleep(self.min_proc_restart_delay)
+        event.clear()
+
+        LOGGER.info("Waiting for threads to join")
+        thread1.join()
+        thread2.join()
+
+        LOGGER.info("Step 4: Checking cluster status")
+        resp = self.ha_obj.poll_cluster_status(self.node_master_list[0], timeout=180)
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Step 4: Cluster status is good")
+
+        LOGGER.info("Step 5: Verifying responses from background processes")
+        LOGGER.info("Checking background process for IAM user CRUDs")
+        iam_resp = tuple()
+        while len(iam_resp) != 4:
+            iam_resp = iam_output.get(timeout=HA_CFG["common_params"]["60sec_delay"])
+        if not iam_resp:
+            assert_utils.assert_true(False, "Background process failed to do IAM user CRUD "
+                                            "operations")
+        exp_fail = iam_resp[0]
+        failed = iam_resp[1]
+        user_del_failed = iam_resp[2]
+        created_users = iam_resp[3]
+        if failed:
+            assert_utils.assert_true(False, "Failures observed in background process for IAM "
+                                            f"user CRUD operations. \nFailed buckets: {failed}")
+        elif exp_fail:
+            LOGGER.info("In-Flight IAM user creation/deletion failed for users: %s", exp_fail)
+            LOGGER.info("In-Flight IAM user deletion failed for users: %s", user_del_failed)
+            for i_i in user_del_failed:
+                self.s3_clean.update({i_i: del_users[i_i]})
+                self.uids.append(del_users[i_i]["user_name"])
+        else:
+            assert_utils.assert_true(False, "IAM user CRUD operations are expected to be failed "
```

Comment:
Is it MAYBE? IF the request goes to the pod which is not down? OR as we are killing all we are expecting at least some failures?

Reply:
Yes, i will update the loggers.